packages/adapters-library/src/adapters/metamask/products/pooled-staking/metamaskPooledStakingAdapter.ts

```diff
@@ -0,0 +1,144 @@
+import { getAddress } from 'ethers'
+import { AdaptersController } from '../../../../core/adaptersController'
+import { Chain } from '../../../../core/constants/chains'
+import { CacheToDb } from '../../../../core/decorators/cacheToDb'
+import { Helpers } from '../../../../core/helpers'
+import { CustomJsonRpcProvider } from '../../../../core/provider/CustomJsonRpcProvider'
+import {
+  IProtocolAdapter,
+  ProtocolToken,
+} from '../../../../types/IProtocolAdapter'
+import {
+  AdapterSettings,
+  GetPositionsInput,
+  PositionType,
+  ProtocolAdapterParams,
+  ProtocolDetails,
+  ProtocolPosition,
+  TokenType,
+  UnwrapExchangeRate,
+  UnwrapInput,
+} from '../../../../types/adapter'
+import { Protocol } from '../../../protocols'
+import { PoolStaking, PoolStaking__factory } from '../../contracts'
+
+const METAMASK_POOLED_STAKING_CONTRACT = getAddress(
+  '0x4FEF9D741011476750A243aC70b9789a63dd47Df',
+)
+const LIDO_STAKED_ETH = getAddress('0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84')
+
+export class MetamaskPooledStakingAdapter implements IProtocolAdapter {
+  productId = 'pooled-staking'
+  protocolId: Protocol
+  chainId: Chain
+  helpers: Helpers
+
+  adapterSettings: AdapterSettings = {
+    includeInUnwrap: true,
+    userEvent: {
+      topic0:
+        '0x861a4138e41fb21c121a7dbb1053df465c837fc77380cc7226189a662281be2c',
+      userAddressIndex: 2,
```

Comment:
I changed this to index 2.

Index 1 is the caller, that is not necessarily who gets the position, which is the receiver in index 2.